### PR TITLE
feat(EFB/SimBrief): Automatically import SimBrief Fuel & Payload (2nd Attempt)

### LINKED ISSUE #7772: Auto-import simbrief data for GSX integration
### Aircraft Version

Experimental

### Description

Now having the gsx integration could you consider adding an option to automatically load simbrief payload data? Similarly to the one we have for dashboard page currently? As for the refuellig ideally would be to import simbrief fuel and start refueling once the hose connected (there is a gsx lvar). The idea is to minimize steps interacting with efb and automate the process. So you spawn at gate, turn on the efb, request boarding and refuelling via gsx, and everything happens on its own.

### References (optional)

_No response_

### Additional info (optional)

_No response_

### Discord Username (optional)

K8#4795

### PR DESCRIPTION
**Pull request by Fragtality**
Fixes #7772
(Together with PR #8407)

## Summary of Changes

New in this Variant:
- Don't call fetchData when simbriefDataPending or both navigraphUsername & overrideSimBriefUserID are empty (although autoSimbriefImport is ENABLED)
- history.push Calls consolidated into one .then Block (and not called when Exception was raised)
- Toast only really displayed when both fuelImported and payloadImported are true
  - Additional Flag in SimBrief Store to show the Notification only once

EDIT: Changes made by @Maximilian-Reuter :
- generally disabled input on fuel and payload pages for changing the desired values while the respective processes are ongoing. 
- Simbrief import is only possible if no fuel or boarding action is ongoing to not affect those actions
- removed the (in my opinion) redundant import button on the OFP page
- simplified the button html on the dashboard
- made sure simbrief api calls are only made when a username or id is supplied

Old Summary:
- When SimBrief Data was imported (either manual or automatic), the Fuel & Payload Pages automatically import their Data too. Approach:
  - The Fuel and Payload Page have an useEffect to import the Data when visited (only when SimBrief Data is loaded)
  - The Import Function navigates through the Pages (Dashboard->Fuel->Payload->Dashboard)
  - Two additional Flags in the SimBrief Store prevent that Payload or Fuel are imported every time the respective Page is revisited
- Auto-Import of SimBrief Data moved from EFB.tsx to FlightWidget.tsx (to not maintain two Copies of the same Code)

## Screenshots (if necessary)

flightwidget-toast

## Additional context

Although the originating Issue is in the GSX Context, this Feature improves Comfort for everyone (using SimBrief). Only one Click needed - if at all - to get all relevant Data imported 😃 

Discord username (if different from GitHub): Fragtality

## Testing instructions

PREPARATION before installing this Build/Artifact:
1. Load into the Airplane with the current/official Dev Version
2. Unlink Navigraph (Settings > 3rd Party Options)
3. Remove SimBrief ID (Settings > 3rd Party Options)
4. Keep the automatic SimBrief Data Import enabled! (Settings > ATSU/AOC)

Test 1

1. Load into the Airplane
2. Power up the EFB => Result: No Error or Success Toasts/Messages
3. Hit "Import SimBrief Data" => Result: Only "No username or id supplied!" should show up
4. Preparation for next Test: set SimBrief ID (Settings > 3rd Party Options) and exit to Main Menu

Test 2

1. Prepare a Flightplan on SimBrief for the FBW
2. Load into the Airplane
3. Power up the EFB => Result: A Notification should appear (like shown in the Screenshots)
4. Navigate to the Fuel & Payload Page => Result: Each Page should display the planned Fuel / Payload from your SimBrief OFP
5. Navigate back to Dashboard => Result: No Notifications shown
6. Edit the SimBrief OFP with a different ZFW & Block-Fuel
7. Click the SimBrief Import Button on the Dashboard => Result: The Notification should appear again
8. Navigate to the Fuel & Payload Page => Result: Each Page should display the new planned Fuel / Payload
9. Preparation for next Test: disable the automatic SimBrief Data Import (Settings > ATSU/AOC) and exit to Main Menu

Test 3

1. Load into the Airplane
2. Power up the EFB => Result: No Error or Success Toasts/Messages and no Import happening
3. Hit "Import SimBrief Data" => Result: OFP imported and Notification for Fuel & Payload is shown
4. Navigate to the Fuel & Payload Page => Result: Each Page should display the new planned Fuel / Payload

Test 4

1. Load up the Airplane / or simply restart the EFB using the menu at the top right (powersymbol)
2. No Import should happen
3. Start a fueling process
4. try to interact with the slider or input field -> shouldnt work
5. stop fueling process 
6. interactions should work again
7. start a boarding process on the payload page
8. nothing but the stop button should work, clicking the seatmap shouldnt do anything
9. stop boarding process 
10. interactions should work as before

Test 5

1. Load up the Airplane / or simply restart the EFB using the menu at the top right (powersymbol)
2. No Import should happen
3. Start a fueling process
4. try to import simbrief on the dashboard
5. Notification should appear :No import possible due to boarding or refuel in progress!
6. stop fueling process 
8. start a boarding process on the payload page
9. try to import simbrief on the dashboard
10. Notification should appear :No import possible due to boarding or refuel in progress!
11. stop boarding process 
12. try to import simbrief on the dashboard
13. Import should work and notification should appear for fuel and payload

GSX specific testing:

This means that under 3rd party settings in the EFB the GSX payload and fuel sync needs to be set.

Test 6

1. Load up the Airplane / or simply restart the EFB using the menu at the top right (powersymbol)
2. No Import should happen
3. Request refueling from GSX
4. try to interact with the slider or input field -> should work until fuel hose is connected
5. stop fueling process 
6. interactions should work again
7. Request boarding from GSX
8. nothing but the stop button should work, clicking the seatmap shouldnt do anything
9. either wait or abort boarding via GSX
10. interactions should work as before

Test 7

1. Load up the Airplane / or simply restart the EFB using the menu at the top right (powersymbol)
2. No Import should happen
3. Request refueling from GSX
4. try to import simbrief on the dashboard-> should work until fuel hose is connected otherwise Notification should appear: No import possible due to boarding or refuel in progress!
5. stop fueling process 
6. interactions should work again
7. Request boarding from GSX
8. try to import simbrief on the dashboard
9. Notification should appear: No import possible due to boarding or refuel in progress!
10. either wait or abort boarding via GSX
11. try to import simbrief on the dashboard
12. Import should work and notification should appear for fuel and payload

For additional Verification you can monitor the L-Vars `A32NX_FUEL_DESIRED` and `A32NX_AIRFRAME_ZFW_DESIRED` => these should immediately Change when the SimBrief Data was imported (without leaving the Dashboard).
(If you have a Tool at Hand to monitor L-Vars quick & easy)

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Comment by Fragtality** [comment]
This PR would be ready to Review / ready for Feedback. It's just in Draft in Case of a Review / Feedback would require bigger Changes / Refactoring

**Comment by 2hwk** [comment]
``fbw-a32nx`` -> ``fbw-common`` as of #8473 being merged

**Comment by Fragtality** [comment]
. . .

File: fbw-common/src/systems/instruments/src/EFB/Ground/Pages/Payload/A320_251N/A320Payload.tsx [hunk 1]
```diff
@@ -140,10 +140,12 @@ export const A320Payload: React.FC<A320Props> = ({
         COMPLETED: 6,
     };
 
+    const generalBoardingInProgress = () => (gsxPayloadSyncEnabled ? gsxInProgress() : boardingStarted); // TODO this should be a Lvar and written in rust
```
Comment (2hwk) [inline fbw-common/src/systems/instruments/src/EFB/Ground/Pages/Payload/A320_251N/A320Payload.tsx:143]: Remove the TODO or reexamine this general boarding var.

On review now that I've had a better look I really don't like this, the rust backend has been made with the concept that GSX has its own seperate hook via the payload layer to the MSFS payload values, we don't manage this at all.

The addition of the GSX vars here was always a temporary solution, meant to be replaced by a dedicated GSX frontend. (See https://github.com/flybywiresim/aircraft/pull/8225)

**Review by 2hwk — CHANGES_REQUESTED** [review]

**Review by 2hwk — APPROVED** [review]

**Comment by Maximilian-Reuter** [comment]
I still need to update the Test descriptions so I would refrain from testing until I have done that

**Comment by alepouna** [comment]
QA Report

Discord: alepouna
Object of testing:  #8470
Tier of Testing : 1
Date : 2/3/2024

Testing Process:
As described
- [PASS] TEST 1 
- [PASS] TEST 2
- [PASS] TEST 3
- [PASS] TEST 4
- [PASS] TEST 5

-- GSX TESTS --

- [PASS] TEST 6
- [PASS] TEST 7

Testing Results:
Passed

Note: 
GSX was installed.

**Comment by alepouna** [comment]
QA Report

Discord: alepouna
Object of testing:  #8470
Tier of Testing : 2
Date: 3/3/2024

Testing Process:
As described, performed full flight. 
- [PASS] TEST 1 
- [PASS] TEST 2
- [PASS] TEST 3
- [PASS] TEST 4
- [PASS] TEST 5

Testing Results:
Passed

Note: 
GSX was NOT installed.

**Pull request merged**